HCollections.Test/TestHList.fs

```diff
@@ -36,3 +36,33 @@ module TestHList =
             |> TypeList.cons<float, _>
 
         Assert.Equal<Type list> (TypeList.toTypes expected, HList.toTypeList hlist |> TypeList.toTypes)
+
+    [<Fact>]
+    let ``HList.split on an empty HList returns the proof that it's an empty HList`` () =
+        let testHList =
+            HList.empty
+        let result = HList.split testHList
+        match result with
+        | Choice1Of2 t -> Teq.castFrom t ()
+        | Choice2Of2 _ -> failwith "The HList is not empty."
+
+    [<Fact>]
+    let ``HList.split on a non-empty HList returns the elements at the head of the HList and the tail`` () =
+        let emptyHList =
+            HList.empty
+        let testHList =
+            emptyHList
+            |> HList.cons "hi"
+        let result = HList.split testHList
+        match result with
+        | Choice1Of2 _ -> failwith "The HListT is empty."
+        | Choice2Of2 c ->
+            c.Apply
+                { new HListConsEvaluator<string -> unit,int> with
+                    member __.Eval (t,head,tail) =
+                        let head = Teq.castFrom (Teq.Cong.domainOf t) head
+                        let tail = Teq.castFrom (HList.cong (Teq.Cong.rangeOf t)) tail
+                        Assert.Equal<_> (emptyHList, tail)
+                        Assert.Equal (head, "hi")
+                        5
```

Comment:
Is this `FakeUnit` in a different guise? ;) Happy for this to stay if so, it's just very sad

Reply:
You are correct! Can make a fake unit type if that's better

Reply:
I have only an extremely mild preference for it.

Reply:
You could probably use Teqs to return the head and tail from this method and have the asserts out the crate. Then no need for fake returns.

Reply:
You are correct, I've changed the tests so we can assert outside the crate.